src/AdoPat/PatOptions.cs

```diff
@@ -33,9 +36,23 @@ public record PatOptions
         /// <returns>The cache key.</returns>
         public string CacheKey()
         {
-            var strings = new List<string> { this.Organization, this.DisplayName };
-            strings.AddRange(this.Scopes);
-            return string.Join(" ", strings);
+            // We concatenate scopes with the empty string after sorting and
+            // deduplicating them to ensure the cache key is always the same.
+            var sortedScopes = string.Concat(this.Scopes.ToImmutableSortedSet());

```

Comment:
Will this take care of case sensitivity as well?

Reply:
We should write a test for checking that as well

Reply:
This is also a great scope _validation_ question, and I'd love to see it handled in a follow up PR.

Reply:
It won't, but I think we should. Like with my comment to @shalinikhare27 earlier, I think this is probably validation that belongs at an early stage in the process. I think it's a good idea to do, but maybe not here in the code.

Reply:
By scope validation do you mean we will not allow `Test.scope.a` or that we will convert that to lowercase there in early stage process?

Reply:
Either way, resolving this as we will be doing that in follow -up PR.